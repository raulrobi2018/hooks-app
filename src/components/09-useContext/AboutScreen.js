import React, {useContext} from "react";
import {UserContext} from "./UserContext";

export const AboutScreen = () => {
    const {user, setUser} = useContext(UserContext);

    const logout = () => {
        setUser({});
    };

    return (
        <div>
            <h1>About</h1>
            <hr />

            <pre>{JSON.stringify(user, null, 3)}</pre>

            <button className="btn btn-secondary" onClick={logout}>
                Logout
            </button>
        </div>
    );
};
