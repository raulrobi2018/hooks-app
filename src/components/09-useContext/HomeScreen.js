import React, {useContext} from "react";
import {UserContext} from "./UserContext";

export const HomeScreen = () => {
    const userContext = useContext(UserContext);

    console.log(userContext);

    return (
        <div>
            <h1>Home</h1>
            <hr />
        </div>
    );
};
