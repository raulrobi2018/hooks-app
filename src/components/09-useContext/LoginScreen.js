import React, {useContext} from "react";
import {UserContext} from "./UserContext";

export const LoginScreen = () => {
    const {setUser} = useContext(UserContext);

    return (
        <div>
            <h1>Login</h1>
            <hr />

            <button
                className="btn btn-primary"
                onClick={() =>
                    setUser({
                        id: 1,
                        name: "Raul Rodriguez",
                        age: 42,
                        email: "raulrobi@gmail.com"
                    })
                }
            >
                Login
            </button>
        </div>
    );
};
