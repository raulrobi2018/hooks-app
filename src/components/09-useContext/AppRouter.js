import React from "react";
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";
import {AboutScreen} from "./AboutScreen";
import {HomeScreen} from "./HomeScreen";
import {LoginScreen} from "./LoginScreen";
import {NavBar} from "./NavBar";

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <Switch>
                    <Route exact path="/" component={HomeScreen} />
                    <Route exact path="/about" component={AboutScreen} />
                    <Route exact path="/login" component={LoginScreen} />

                    {/* Si ingresan una ruta que no existe, entra aquí y redirecciona a / (barra).
                     También se puede crear una página 404 y poner otro Route al final que direccione
                     a esa página */}
                    <Redirect to="/" />
                </Switch>
            </div>
        </Router>
    );
};
