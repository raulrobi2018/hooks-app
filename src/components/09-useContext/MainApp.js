import React from "react";
import {AppRouter} from "./AppRouter";
import {UserContext} from "./UserContext";

export const MainApp = () => {
    const user = {
        id: 1,
        name: "Raul",
        age: 42,
        email: "raulrobi@gmail.com"
    };

    return (
        <UserContext.Provider value={user}>
            <AppRouter />
        </UserContext.Provider>
    );
};
