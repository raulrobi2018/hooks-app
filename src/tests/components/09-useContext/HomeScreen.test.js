import {mount, configure} from "enzyme";
import {HomeScreen} from "../../../components/09-useContext/HomeScreen";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import {UserContext} from "../../../components/09-useContext/UserContext";

//ConfiguraciÃ³n necesaria para que funcione mount con React 17
configure({adapter: new Adapter()});

describe("Testing HomeScreen", () => {
    const user = {
        name: "Raul",
        email: "rr@gmail.com"
    };
    const wrapper = mount(
        <UserContext.Provider value={{user}}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test("should display the component correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });
});
