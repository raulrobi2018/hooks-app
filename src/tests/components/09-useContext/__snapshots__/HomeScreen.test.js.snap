// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomeScreen should display the component correctly 1`] = `
<div>
  <h1>
    Home
  </h1>
  <hr />
  <pre>
    {
   "name": "Raul",
   "email": "rr@gmail.com"
}
  </pre>
</div>
`;
